### PR TITLE
feat: e-mail templates

### PR DESCRIPTION
**Pull request by michaelulrichwaeger**
Hallo mitenand 
Könnt Ihr bitte die Cusomized Notifications prüfen. Danke.

### CODE REVIEW

File: internal/notification/types/domain_claimed.go [hunk 1]
```diff
@@ -27,11 +27,28 @@ func SendDomainClaimed(dir http.FileSystem, i18n *i18n.Translator, user *view_mo
...
 	}
-	systemDefaults.Notifications.TemplateData.DomainClaimed.Translate(i18n, args, user.PreferredLanguage)
-	data := &DomainClaimedData{TemplateData: systemDefaults.Notifications.TemplateData.DomainClaimed, URL: url}
-	template, err := templates.GetParsedTemplate(dir, data)
+
+	text.Greeting, err = templates.ParseTemplateText(text.Greeting, args)
+	text.Text, err = templates.ParseTemplateText(text.Text, args)
+	text.Text = html.UnescapeString(text.Text)
+
+	emailCodeData := &EmailVerificationCodeData{
```
Comment (fgerschwiler) [inline internal/notification/types/domain_claimed.go:35]: Shouldnt this be DomainClaimedData?

File: internal/notification/types/init_code.go [hunk 2]
```diff
@@ -38,15 +37,28 @@ func SendUserInitCode(dir http.FileSystem, i18n *i18n.Translator, user *view_mod
...
 
-	// Set the color in initCodeData
-	initCodeData.PrimaryColor = colors.PrimaryColor
-	initCodeData.SecondaryColor = colors.SecondaryColor
-	template, err := templates.GetParsedTemplate(dir, initCodeData)
+	text.Greeting, err = templates.ParseTemplateText(text.Greeting, args)
+	text.Text, err = templates.ParseTemplateText(text.Text, args)
+	text.Text = html.UnescapeString(text.Text)
+
+	emailCodeData := &EmailVerificationCodeData{
```
Comment (fgerschwiler) [inline internal/notification/types/init_code.go:45]: Shouldn this be InitCodeEmailData?

File: internal/notification/types/password_code.go [hunk 3]
```diff
@@ -33,15 +32,28 @@ func SendPasswordCode(dir http.FileSystem, i18n *i18n.Translator, user *view_mod
...
 
-	// Set the color in initCodeData
-	passwordCodeData.PrimaryColor = colors.PrimaryColor
-	passwordCodeData.SecondaryColor = colors.SecondaryColor
-	template, err := templates.GetParsedTemplate(dir, passwordCodeData)
+	text.Greeting, err = templates.ParseTemplateText(text.Greeting, args)
+	text.Text, err = templates.ParseTemplateText(text.Text, args)
+	text.Text = html.UnescapeString(text.Text)
+
+	emailCodeData := &EmailVerificationCodeData{
```
Comment (fgerschwiler) [inline internal/notification/types/password_code.go:40]: Shouldnt this be PasswordCodeData?

File: internal/org/repository/eventsourcing/label_policy_test.go [hunk 4]
```diff
@@ -30,7 +30,7 @@ func TestLabelPolicyAddedAggregate(t *testing.T) {
 		res  res
 	}{
 		{
-			name: "add label polciy",
+			name: "add label policiy",
```
Comment (fgerschwiler) [inline internal/org/repository/eventsourcing/label_policy_test.go:33]:
```suggestion
			name: "add label policy",
```

**Review by fgerschwiler — COMMENTED** [review]
i think some error messages are missing in de translation files (de.yaml & en.yaml) eg: MailTemplate missing

File: internal/iam/repository/view/mail_template_view.go [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+	iam_model "github.com/caos/zitadel/internal/iam/model"
+	"github.com/caos/zitadel/internal/iam/repository/view/model"
+	global_model "github.com/caos/zitadel/internal/model"
+	"github.com/caos/zitadel/internal/view/repository"
+	"github.com/jinzhu/gorm"
+)
+
+func GetMailTemplateByAggregateID(db *gorm.DB, table, aggregateID string) (*model.MailTemplateView, error) {
+	template := new(model.MailTemplateView)
+	userIDQuery := &model.MailTemplateSearchQuery{Key: iam_model.MailTemplateSearchKeyAggregateID, Value: aggregateID, Method: global_model.SearchMethodEquals}
```
[the remark this answers is not in the document]
Comment (michaelulrichwaeger) [reply internal/iam/repository/view/mail_template_view.go:14]: Should I correct this in all files?
image

[the remark this answers is not in the document]
Comment (fgerschwiler) [reply internal/iam/repository/view/mail_template_view.go:14]: yes this would be perfect

**Review by fgerschwiler — APPROVED** [review]

**Pull request merged**